static/app/utils/marked/marked.tsx

```diff
@@ -0,0 +1,144 @@
+import dompurify from 'dompurify';
+import type {Tokens} from 'marked'; // eslint-disable-line no-restricted-imports
+import {Marked, marked} from 'marked'; // eslint-disable-line no-restricted-imports
+import {markedHighlight} from 'marked-highlight';
+import Prism from 'prismjs';
+
+import {loadPrismLanguage} from 'sentry/utils/prism';
+
+// Only https and mailto, (e.g. no javascript, vbscript, data protocols)
+const safeLinkPattern = /^(https?:|mailto:)/i;
+
+const safeImagePattern = /^https?:\/\/./i;
+
+function isSafeHref(href: string, pattern: RegExp) {
+  try {
+    return pattern.test(decodeURIComponent(unescape(href)));
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Implementation of marked. Renderer which additionally sanitizes URLs.
+ */
+class SafeRenderer extends marked.Renderer {
+  link({href, title, text, ...rest}: Tokens.Link) {
+    // For a bad link, just return the plain text href
+    if (!isSafeHref(href, safeLinkPattern)) {
+      return href;
+    }
+
+    const out = super.link({href, title, text, ...rest});
+    return dompurify.sanitize(out);
+  }
```

Comment:
Not a blocker on this PR, but what are your thoughts on adding in `rel="noopener noreferrer"` on some links? It'd only be on input where the link has `target="_blank"` set and doesn't have the `rel` tag (or does, but no `noopener noreferrer` in the list? 

Could help us prevent spots where tab-nabbing may be possible.

Reply:
So it's not possible for the links to have _target from my knowledge. we would strip any A tags from the input and any generated by the markdown would not have any attributes.